src/Microsoft.VisualStudio.ProjectSystem.Managed.VS.UnitTests/ProjectSystem/VS/Utilities/StringExtensions.cs

```diff
@@ -0,0 +1,27 @@
+﻿using System;
+using System.IO;
+using System.Xml;
+using Microsoft.Build.Construction;
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS.Utilities
+{
+    public static class StringExtensions
+    {
+        public static ProjectRootElement AsProjectRootElement(this string @string)
+        {
+            var stringReader = new System.IO.StringReader(@string);
+            var xmlReader = new XmlTextReader(stringReader);
+            var root = ProjectRootElement.Create(xmlReader);
+            return root;
+        }
+
+        public static string SaveAndGetChanges(this ProjectRootElement root)
+        {
+            var tempFile = Path.GetTempFileName();
```

Comment:
@vshapenko saving to a string will cause a different code path to be used that writes out the file differently. I am writing to disk here because I want the same behavior in the unit tests and what MSBuild actually does. I can look for other ways around this and revisit this is a subsequent PR